docs/pages/config-plugins/introduction.mdx

```diff
@@ -1,59 +1,62 @@
 
-Expo config plugins mostly come from Node.js modules. You can install them just like other packages in your project.
+## Mods
 
-For example, `expo-camera` has a plugin that adds camera permissions to the **AndroidManifest.xml** and **Info.plist**. To install it in your project, run the following command:
+Config plugins use **mods** (short for modifiers) to modify native project files during the prebuild process. Mods are asynchronous functions that allow you to make changes to platform-specific files such as **AndroidManifest.xml** and **Info.plist**, and other native configuration files without having to manually edit them. They accept a config and a data object, then modify and return both them as a single object. For example, in native projects, `mods.android.manifest` modifies **AndroidManifest.xml** and `mods.ios.plist` modifies **Info.plist**
 
-<Terminal cmd={['$ npx expo install expo-camera']} />
+**You don't have to use mods as top-level functions (for example `with.android.manifest`) directly in your config plugin.** When you need to use a mod, you use _mod plugins_ in your config plugins. These mod plugins are provided by the `expo/config-plugins` library and wrap top-level mod functions and behind the scenes they perform various tasks. To see a list of available mods, check out the [mod plugins provided by `expo/config-plugins`](/config-plugins/mods/#mod-plugins).
 
-In your [app's config](/versions/latest/config/app/), you can add `expo-camera` to the list of plugins:
+## Characteristics of a plugin
 
-```json app.json
-{
-  "expo": {
-    "plugins": ["expo-camera"]
-  }
-}
-```
+The plugin functions have the following characteristics:
 
-Some config plugins offer flexibility by allowing you to pass options to customize their configuration. To do this, you can pass an array with the Expo library name as the first argument, and an object containing the options as the second argument. For example, the `expo-camera` plugin allows you to customize the camera permission message:
+- Plugins are **synchronous** functions that accept an [ExpoConfig](/workflow/configuration) and return a modified `ExpoConfig`. In rare cases, plugins can also be asynchronous if available methods to communicate with native projects are asynchronous, but they won't be performant.
+- Plugins should be named using the following convention: `with<Plugin Functionality>`, for example, `withFacebook`
+- Plugins should be synchronous and their return value should be serializable, except for adding any [`mods`](#mods)
+- Optionally, a second argument can be passed to the plugin to configure it
+- Mods are only invoked during the **syncing** phase of `npx expo prebuild` (prebuild process) and modify native files during code generation. Because of that, any modifications made to app config in a config plugin should be outside of a mod to ensure that they are executed in non-prebuild configuration scenarios.
```

Comment:
This is worth unpacking a little more: that plugins are _always_ evaluated during config evaluation, but mods only execute when prebuilding the platform that the mod is for. Might describe this in the "mods" section.

Reply:
Since you mentioned this page should have a glossary section and all details regarding mods should go on the mods page, I'll add this context there.

For plugins, I've also added an item about plugins evaluation describing what you mentioned in your comment above.